lesson-7/server/test/api.js

```diff
@@ -0,0 +1,105 @@
+const chai = require('chai');
+const api = require('../api');
+const config = require('../config');
+const collection = 'todos_TEST'; // use separate collection for tests
+
+const testItems = [
+    {
+        _id: 1,
+        text: 'test1'
+    },
+    {
+        _id: 2,
+        text: 'test2'
+    },
+    {
+        _id: 3,
+        text: 'test3'
+    }
+];
+
+describe('todo API Tests', () => {
+    before((done) => {
```

Comment:
This can be simpler thanks to Promises:

```javascript
before(() => {
    // initialize the api
    return api.initConnectionPool(config.MONGO_URL, collection)
    .then(() => api.resetCollection().catch(error => { console.log(error) }))
    .then(() => 
        // create test items
        Promise.all(testItems.map((item) => api.createTodoItem(item._id, item)))
    )
});
```

Same for all other tests here